### DISCUSSION TITLE
[V8] Vue 3 support

### DISCUSSION THREAD

**Discussion by hanneskuettner** [TanStack/table#3109] [General] [answered]
Hi,

looks like V8 is [well on its way](https://github.com/tannerlinsley/react-table/discussions/3054) and from what I can piece together it will be [written in Typescript](https://github.com/tannerlinsley/react-table/issues/1591#issuecomment-729945218). 🚀 

I was about to attempt a port of react-table to Vue since I'm loving the headless nature of this library and there's nothing like it out there. But then I stumbled upon plans or at least ideas to support Vue 3 in other TanStack projects like [react-query](https://github.com/tannerlinsley/react-query/discussions/637).
Looking at the discussion over at react-query it seems fairly straightforward to find equivalents for react hooks over in the Vue 3 composition API.

Now I wonder if Vue 3 support is considered or at least made easier by separating core and react specific implementations?

**Comment by tannerlinsley** [comment, marked as answer]
React Table will similarly be built with a separate internal core module. It will be even easier to use with Vue than react query.

**Reply by hanneskuettner** [reply]
Wonderful news! Very much looking forward to the first publicly available version.

**Reply by Irev-Dev** [reply]
Expressing my interest in a Vue implemention! 🙌

**Reply by raphaelsaunier** [reply]
Great news! In the meantime, what are you folks using in the Vue ecosystem? I had a quick look and nothing seems to come close to react-table… few libraries use the Vue 3 composition API and they're all tightly coupled and too heavily styled to my taste.

**Reply by Gravy59** [reply]
What should we use for TypeScript in the meantime?

**Reply by vincerubinetti** [reply]
It looks like, as of a few days ago, v8 (core, react, vue, etc) alpha is available, wow! 😍 `@tanstack/vue-table@8.0.0-alpha.87`
https://github.com/TanStack/react-table/releases/tag/v8.0.0-alpha.87

Just curious @tannerlinsley . It looks like v8 is the first version to support frameworks other than React? I know you already say this in the readme:

<img width="917" alt="Screen Shot 2022-05-25 at 12 58 11 PM" src="https://user-images.githubusercontent.com/8326331/170319261-674f2e38-acdc-4295-b1cd-3e631189999a.png">

but you might want to reallllly highlight that this supports Vue and other libraries, like big and bold, since the name of the library says "React", and I'm also kinda used to your libraries only being for React. Like I'd even put `(and Vue, Svelte, and more)` in the [logo image](https://raw.githubusercontent.com/tanstack/react-table/alpha/media/repo-dark.png).

I've spent the last week scouring the internet for an acceptable Vue 3 table solution, and being sad that I can't find one, and I didn't think to check this library until now. The way this library is headless and modular seems to be the correct way to do things, imo, and I'm shocked there's no Vue equivalent yet.

Perhaps I'm completely misunderstanding all of this, apologies if I'm confused.

**Reply by vincerubinetti** [reply]
Ah, I suppose you're kinda in the process of rebranding it to `TanStack Table`: https://tanstack.com/table/v8

Hopefully that eventually gets reflected in the main GitHub repo containing all the packages (the logo, the readme, etc.), as that's where many Google searches end up.
